File: TP7/ws_i_3_client.js
const WebSocket = require('ws');

const uri = "ws://10.0.3.17:8888";
const socket = new WebSocket(uri);

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Choisissez un pseudo : ', (pseudo) => {
    socket.send(pseudo);

    rl.on('line', (userInput) => {
        if (userInput.toLowerCase() === 'exit') {
            rl.close();
        } else {
            socket.send(userInput);
        }
    });
});

socket.on('message', (message) => {
    console.log(message);
});

socket.on('close', () => {
    console.log("Connexion fermée par le serveur.");
    rl.close();
});

File: TP7/ws_i_3_client.py
import asyncio
import websockets

async def user_interaction():
    uri = "ws://10.0.3.17:8888"
    async with websockets.connect(uri) as websocket:
        pseudo = input("Choisissez un pseudo : ")
        await websocket.send(pseudo)

        try:
            while True:
                user_input = input("Saisissez un message (ou 'exit' pour quitter) : ")

                if user_input.lower() == 'exit':
                    break

                await websocket.send(user_input)

        except websockets.exceptions.ConnectionClosedOK:
            print("Connexion fermée par le serveur.")

if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(user_interaction())
